src/lib.rs

```diff
@@ -78,5 +80,5 @@ pub trait Runtime: std::fmt::Debug {
     fn register(&mut self, name: &str, content: &str) -> std::io::Result<()>;
 
     /// Get the output from typing `input` into the shell
-    fn complete(&mut self, input: &str, term: &Term) -> std::io::Result<String>;
+    fn complete(&mut self, input: &str, term: &Term, timout: Duration) -> std::io::Result<String>;
```

Comment:
It seems like whether a `timeout` is needed or not is dependent on the needs of the individual runtime.  Should this be a function on them instead?